Okta.Core/OktaHttpClient.cs

```diff
@@ -175,6 +176,24 @@ public override Task<HttpResponseMessage> ExecuteAsync(HttpRequestType requestTy
                     {
                         httpClient.DefaultRequestHeaders.Remove("Authorization");
                     }
+                    if (bAddXForwardedHeader)
+                    {
+                        string ipAddress = null;
+                        if (!string.IsNullOrEmpty(content))
+                        {
+                            content = content.Trim();
+                            if (content.StartsWith("{") && content.EndsWith("}"))
+                            {
+                                var obj = JObject.Parse(content);
+                                ipAddress = (string)obj["context"]["ipAddress"];
+                            }
+                        }
+
+                        if (ipAddress != null)
+                        {
+                            httpClient.DefaultRequestHeaders.Add("X-Forwarded-For", ipAddress);

```

Comment:
Sure - we should not set `DefaultRequestHeaders`. Instead, we need to create a `new HttpRequestMessage` and set the header on that instance only. I can give you an example if you aren't familiar with the message object.